### PR TITLE
Update FXIOS-4475 [v106] recently saved with new designs

### PR DESCRIPTION
**Pull request by yoanarios**
Issue #11151
Jira [4475](https://mozilla-hub.atlassian.net/browse/FXIOS-4475)

Includes fix for [3553](https://mozilla-hub.atlassian.net/browse/FXIOS-3553) Issue #9687

### CODE REVIEW

File: Client/Frontend/Home/RecentlySaved/RecentlySavedCell.swift [hunk 1]
```diff
@@ -58,45 +89,113 @@ class RecentlySavedCell: UICollectionViewCell, ReusableCell, NotificationThemeab
...
+        fallbackFaviconImage.image = nil
         itemTitle.text = nil
+        setFallBackFaviconVisibility(isHidden: false)
         applyTheme()
     }
 
+    func configure(site: Site, heroImage: UIImage?, favIconImage: UIImage?) {
+        configureImages(heroImage: heroImage, favIconImage: favIconImage)
+
+        itemTitle.text = site.tileURL.shortDisplayString.capitalized // site.title
```
Comment (yoanarios) [inline Client/Frontend/Home/RecentlySaved/RecentlySavedCell.swift:101]: Checking with UX if we should use `shortDispalyString` or site `title`

File: Client/Frontend/Home/RecentlySaved/RecentlySavedDataAdaptor.swift [hunk 2]
```diff
@@ -63,6 +71,19 @@ class RecentlySavedDataAdaptorImplementation: RecentlySavedDataAdaptor, Notifiab
         return nil
     }
 
+    func getFaviconImage(forSite site: Site) -> UIImage? {
+        if let heroImage = faviconImages[site.url] {
+            return heroImage
+        }
```
Comment (lmarceau) [inline Client/Frontend/Home/RecentlySaved/RecentlySavedDataAdaptor.swift:77]: Should be this?
```suggestion
        if let faviconImage = faviconImages[site.url] {
            return faviconImage
        }
```

File: Client/Frontend/Home/RecentlySaved/RecentlySavedDataAdaptor.swift [hunk 3]
```diff
@@ -7,6 +7,7 @@ import Storage
 
 protocol RecentlySavedDataAdaptor {
     func getHeroImage(forSite site: Site) -> UIImage?
+    func getFaviconImage(forSite site: Site) -> UIImage?
```
Comment (lmarceau) [inline Client/Frontend/Home/RecentlySaved/RecentlySavedDataAdaptor.swift:10]: There's tests for `getHeroImage` in RecentlySavedDataAdaptorTests, we should add some for `getFaviconImage`

File: Client/Frontend/Home/RecentlySaved/RecentlySavedCell.swift [hunk 4]
```diff
@@ -58,45 +87,113 @@ class RecentlySavedCell: UICollectionViewCell, ReusableCell, NotificationThemeab
...
+            setFallBackFaviconVisibility(isHidden: true)
+            self.heroImage.image = heroImage
+        }
+    }
+
+    private func setFallBackFaviconVisibility(isHidden: Bool) {
+        fallbackFaviconBackground.isHidden = isHidden
+        fallbackFaviconImage.isHidden = isHidden
+
+        self.heroImage.isHidden = !isHidden
```
Comment (lmarceau) [inline Client/Frontend/Home/RecentlySaved/RecentlySavedCell.swift:120]: nit
```suggestion
        heroImage.isHidden = !isHidden
```

File: Client/Frontend/Home/RecentlySaved/RecentlySavedCell.swift [hunk 5]
```diff
@@ -58,45 +87,113 @@ class RecentlySavedCell: UICollectionViewCell, ReusableCell, NotificationThemeab
...
+    private func configureImages(heroImage: UIImage?, favIconImage: UIImage?) {
+        if heroImage == nil {
+            // Sets a small favicon in place of the hero image in case there's no hero image
+            fallbackFaviconImage.image = favIconImage
+        } else if heroImage?.size.width == heroImage?.size.height {
+            // If hero image is a square use it as a favicon
+            fallbackFaviconImage.image = heroImage
+        } else {
+            setFallBackFaviconVisibility(isHidden: true)
+            self.heroImage.image = heroImage
```
Comment (lmarceau) [inline Client/Frontend/Home/RecentlySaved/RecentlySavedCell.swift:112]: nit
```suggestion
            heroImage.image = heroImage
```

**Review by lmarceau — APPROVED** [review]

**Comment by yoanarios** [comment]
Screen Shot 2022-09-01 at 10 04 28 AM

**Pull request merged**